bug: building docs for tags is broken

### Describe the bug

Builidng docs for tags does not complete successfully

### Steps To Reproduce

1. git push a tag
2. https://github.com/kcp-dev/kcp/actions/workflows/docs-gen-and-push.yaml fails

### Expected Behaviour

Success

### Additional Context

Need to add a `mkdir -p`

```
2022-12-13T17:15:18.9821869Z ##[group]Run cd kcp/docs
2022-12-13T17:15:18.9822103Z [36;1mcd kcp/docs[0m
2022-12-13T17:15:18.9822315Z [36;1mcat <<EOF > temp.txt[0m
2022-12-13T17:15:18.9822533Z [36;1m#version-start[0m
2022-12-13T17:15:18.9822749Z [36;1marchived_version = true[0m
2022-12-13T17:15:18.9822983Z [36;1mversion = "refs/tags/v0.10.0"[0m
2022-12-13T17:15:18.9823190Z [36;1mEOF[0m
2022-12-13T17:15:18.9823393Z [36;1med config.toml<<EOF[0m
2022-12-13T17:15:18.9823612Z [36;1m/^#version-start[0m
2022-12-13T17:15:18.9823825Z [36;1m+,/^#version-end/-1d[0m
2022-12-13T17:15:18.9824142Z [36;1m-r !sed -n '/^#version-start/,/^#version-end/p' temp.txt | grep -v '^#'[0m
2022-12-13T17:15:18.9824415Z [36;1mw[0m
2022-12-13T17:15:18.9824638Z [36;1mq[0m
2022-12-13T17:15:18.9824812Z [36;1mEOF[0m
2022-12-13T17:15:18.9825047Z [36;1mmv content/en/main content/en/refs/tags/v0.10.0[0m
2022-12-13T17:15:18.9825280Z [36;1mhugo --minify[0m
2022-12-13T17:15:18.9825524Z [36;1mrsync -av public/refs/tags/v0.10.0 ../../docs[0m
2022-12-13T17:15:18.9879540Z shell: /usr/bin/bash -e {0}
2022-12-13T17:15:18.9879774Z ##[endgroup]
2022-12-13T17:15:18.9933688Z ##[debug]/usr/bin/bash -e /home/runner/work/_temp/9efec89e-07c8-4241-893a-ef538cb90208.sh
2022-12-13T17:15:19.0495270Z 4134
2022-12-13T17:15:19.0496268Z #version-start
2022-12-13T17:15:19.0518683Z 54
2022-12-13T17:15:19.0519349Z 4146
2022-12-13T17:15:19.0533656Z mv: cannot move 'content/en/main' to 'content/en/refs/tags/v0.10.0': No such file or directory
2022-12-13T17:15:19.0548700Z ##[error]Process completed with exit code 1.
```

--- a/.github/workflows/docs-gen-and-push.yaml
+++ b/.github/workflows/docs-gen-and-push.yaml
@@ -102,6 +102,7 @@ jobs:
           w
           q
           EOF
+          mkdir -p content/en/${{ github.ref }}
           mv content/en/main content/en/${{ github.ref }}
           hugo --minify
           rsync -av public/${{ github.ref }} ../../docs

